lang/syn/src/codegen/program/handlers.rs

```diff
@@ -694,6 +694,15 @@ pub fn generate(program: &Program) -> proc_macro2::TokenStream {
             let anchor = &ix.anchor_ident;
             let variant_arm = generate_ix_variant(ix.raw_method.sig.ident.to_string(), &ix.args);
             let ix_name_log = format!("Instruction: {}", ix_name);
+            let ret_type = &ix.returns.ty.to_token_stream();
+            let result_handler = match ret_type.to_string().as_str() {
```

Comment:
how about calling it `maybe_set_return_data` and the other one `maybe_get_return_data` or something more distinct? I got confused a couple times because theyre both called `result_handler`